subsys/net/lib/zperf/zperf_shell.c

```diff
@@ -178,6 +178,137 @@ static int parse_ipv4_addr(const struct shell *sh, char *host, char *port,
 	return 0;
 }
 
+static int zperf_dowload_bind_host(const struct shell *sh,
+				   size_t argc, char *argv[],
+				   struct zperf_download_params *param)
+{
+	struct in6_addr sin6_addr;    /* IPv6 address */
+	struct in_addr sin_addr;      /* IPv4 address */
+	int ret;
+
```

Comment:
Please refactor the code to use
```
`bool net_ipaddr_parse(const char *str, size_t str_len,
		      struct sockaddr *addr);
```
which can parse the a string (with port number) to a sockaddr struct.